README.md

```diff
@@ -4,30 +4,45 @@
 
 ## Obligatory warning
 
-Very much a WORK IN PROGRESS at this point. Not really ready for use.
+Very much a WORK IN PROGRESS at this point. Ready for experimental use
+but expect frequent breaking changes.
 
 ## Credits
 
 This system is heavily inspired by adapton, glimmer, and rustc's query
 system. So credit goes to Eduard-Mihai Burtescu, Matthew Hammer,
 Yehuda Katz, and Michael Woerister.
 
-## Goals
+## Key idea
+
+The key idea of `salsa` is that you define your program as a set
+of **queries**. Queries come in two basic varieties:
+
+- **Inputs**: the base inputs to your system. You can change these
+  whenever you like.
+- **Functions**: pure functions (no side effects) that transform your
+  inputs into other values. The results of queries is memoized to
+  avoid recomputing them a lot. When you make changes to the inputs,
+  we'll figure out (fairly intelligently) when we can re-use these
+  memoized values and when we have to recompute them.
+
+## How to use Salsa in three easy steps
+
+Using salsa is as easy as 1, 2, 3...
```

Comment:
That should have `</sarcasm>` tag perhaps? Unless you already know how it is supposed to work, it's pretty hard to decipher what those macros actually do :)